packages/devtools-reps/src/object-inspector/utils/client.js

```diff
@@ -80,6 +81,17 @@ async function getPrototype(
   return objectClient.getPrototype();
 }
 
+async function getFullText(
+  objectClient: ObjectClient,
+  object: NodeContents,
+) : Promise<?string | {}> {
+  if (typeof objectClient.getString !== "function") {
```

Comment:
I guess it's a hack to treat `WebConsoleClient` as an `ObjectClient` and only care about the `getString` method then:

```javascript
    // packages/devtools-reps/src/launchpad/index.js
    getObjectClient: function (grip) {
      if (isLongString(grip)) {
        // Use `WebConsoleClient.getString`.
        return connection.tabConnection.tabTarget.activeConsole;
      }
```

I thought we could avoid passing a longString specific function and increase the OI's and the loadProperties functions' arity, since all property loaders are owned by `ObjectClient` (`enumEntries`, `enumProperties`, `getPrototype`, ...), but if that's confusing and the current `WebConsoleClient.getString` lacks something we can go with your approach.

Reply:
I think it's a but confusing. The webConsoleClient, as its name indicates, is specific to the webconsole. Here the objectInspector can be used in a variety of tools (debugger, inspector, …), and the way they can get a longString client might differ from the console.
If we go this way and let the responsability of the consumer to branch on longString to provide a specific "ObjectClient", I'd rather call it something else so we can't confuse it with the actual ObjectClient class (e.g. ObjectPropertyFetcher, or something with a better name :D). 

I'm still unsure which way we should go for now. I'd say that letting the ObjectInspector chose which client to create and how to fetch data makes the code easier to reason about and use from a consumer standpoint, whereas if we need to branch on the consumer on specific node types, the ObjectInspector delays some of its responsabilities. 

I'm not trying to impose something here and I'm genuinely hesitating between the two approoaches. What do you think @kootoopas ?

Reply:
I hope I got the functionality and your words right. I will attempt to give some more arguments and a conclusion.

The `ObjectPropertyFetcher` seems more reasonable if we're gonna go with the single fetcher  function (now `getObjectClient`) approach, but to be honest I'd rather delegate picking the right client for the grip to the `OI` since it will make `OI` initialization more straightforward in dependents. Else, if I understand correctly, we're forcing dependents to implement a complicated `getObjectClient` (or `getObjectPropertyFetcher`):
https://github.com/kootoopas/devtools-core/blob/03ebe023945283302de8aa984fc700747f0c7ca2/packages/devtools-reps/src/launchpad/index.js#L35-L42

So basically, I'm voting for the addition of a `createLongStringClient` prop in `ObjectClient` that will result to easier consumer usage! (given that my args are valid 🤞)